From b260c3354d5e9318321c2fc482724870fd9b2740 Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Mon, 23 Jul 2018 15:12:38 -0700
Subject: [PATCH 095/103] parted.c: Always free peek_word

If command_line_get_fs_type failed it would never free it, so put a free
in both branches of the if.

Related: rhbz#1602652
---
 parted/parted.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/parted/parted.c b/parted/parted.c
index 267c346..0dc38c3 100644
--- a/parted/parted.c
+++ b/parted/parted.c
@@ -684,12 +684,13 @@ do_mkpart (PedDevice** dev, PedDisk** diskp)
         if (part_type == PED_PARTITION_EXTENDED
             || (peek_word && (isdigit (peek_word[0]) || peek_word[0] == '-'))) {
                 fs_type = NULL;
+                free (peek_word);
         } else {
+                free (peek_word);
                 if (!command_line_get_fs_type (_("File system type?"),
                                                &fs_type))
                         goto error;
         }
-        free (peek_word);
 
         if (!command_line_get_sector (_("Start?"), *dev, &start, &range_start, NULL))
                 goto error;
-- 
2.17.2

